src/gui/mainwindow.cpp

```diff
@@ -1095,15 +1104,31 @@ void MainWindow::dropEvent(QDropEvent *event)
         files = event->mimeData()->text().split('\n');
     }
 
-    // Add file to download list
+    // differentiate ".torrent" files and others
+    QStringList torrentFiles, otherFiles;
+    foreach (const QString &file, files) {
+        if (file.endsWith(".torrent", Qt::CaseInsensitive))
+            torrentFiles << file;
+        else
+            otherFiles << file;
+    }
+
+    // Download torrents
     const bool useTorrentAdditionDialog = AddNewTorrentDialog::isEnabled();
-    foreach (QString file, files) {
+    foreach (const QString &file, torrentFiles) {
         qDebug("Dropped file %s on download list", qPrintable(file));
         if (useTorrentAdditionDialog)
             AddNewTorrentDialog::show(file, this);
         else
             BitTorrent::Session::instance()->addTorrent(file);
     }
+    if (!torrentFiles.isEmpty()) return;
+
+    // Create torrent
+    foreach (const QString &file, otherFiles) {
```

Comment:
this is a stub that *might* (or not) expand later to create many torrents at once (if there is a option for it)...